### ISSUE TITLE
DOC: remove empty html to reduce size of the artifact

### ISSUE THREAD

**Issue by tupui** [scipy/scipy#15674]
Looking at one of the output [log](https://circleci.com/api/v1.1/project/github/scipy/scipy/43940/output/105/0?file=true&allocation-id=621b2f40232ef87d924b53fa-0-build%2F77A175BB) of the artifact upload, one can notice that there are lot of entries for lapack and blas. But these pages do not bring any value since they are just wrappers to low level functions. See for example https://docs.scipy.org/doc/scipy-1.8.0/html-scipyorg/reference/generated/scipy.linalg.lapack.zungrq.html 

I checked locally and removing these would save:

* Lapack 580 files: 11Mo
* Blas 141 files: 2.7 Mo

On 125Mo, it's 10% for an easy change with no impact for the user. Considering the volume of builds we have, this is not negligible IMO. This would reduce the upload time and our network consumption.

The next contestant for sliming things would be MathJax. Here I don't know so would leave this out for now.

Apart from that, there is not much we can do with upload speed. CircleCI is storing the artifacts on S3 so we cannot upload an archive and unarchive it there. Going further I think we would need to use something like Netlify which is more appropriate for websites.

**Comment by ev-br** [comment]
Would be nice to actually add some docstrings to BLAS/LAPACK functions. That would be quite a job though: starting from netlib sources, one would need to go through all our wrappers and manually adjust signatures for hidden low-level arguments (leading dimensions etc). So maybe not worth the effort, the ROI is too low.

**Comment by ilayn** [comment]
LAPACK funcs actually have some use but as you said it can be smaller if we can find an alternative representations. Because our wrappers are not just wrappers we change the API substantially. Most importantly, chars become ints or bools, e.g., any `job='L'` or `job='N'` becomes `lower=1` or `eigv=0` and so on. Not many customers but still some

**Comment by tupui** [comment]
Ok these are good precisions. Not as simple as removing then, but maybe a single html with the correct signature would do instead (so the current listing pages but without links to individual functions).

**Comment by rgommers** [comment]
> Not many customers but still some .

I guess that's the dilemma. There aren't many users, but they are public API functions and there is real content here:

<img width="485" alt="image" src="https://user-images.githubusercontent.com/98330/158596099-1003d169-3ca9-4f1a-99c7-8b2b5b8439de.png">

It comes from the `f2py` signature files. If we throw that all away, then anyone who needs info should then just reads the raw f2py signature files. 

> Not as simple as removing then, but maybe a single html with the correct signature would do instead (so the current listing pages but without links to individual functions).

That is still throwing away all this information? The size of the html docs is a pain, but this doesn't seem justified. Further optimizing the Sphinx theme seems like a better way to go I'd think.

> as you said it can be smaller if we can find an alternative representations

There is little superfluous info on each page that can be removed. The size issue simply comes from the number of pages. We could reduce it by having multiple functions on a single page, but that'd be a fairly large amount of work I'd think in our Sphinx machinery or `numpydoc`.

**Comment by rgommers** [comment]
Yes, that would make sense. The best example to look at is probably https://github.com/scipy/scipy/blob/main/doc/source/scipyoptdoc.py, which does something custom for the different `optimize.minimize` methods. It's not all that easy to deal with unfortunately.
